src/App.jsx

```diff
@@ -0,0 +1,135 @@
+/********************************************************************************
+ * Copyright (c) 2018 - 2020 Contributors to the Eclipse Foundation
+ * 
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the W3C Software Notice and
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR W3C-20150513
+ ********************************************************************************/
+import React, { useContext, useEffect, useState } from 'react';
+import ediTDorContext from "./context/ediTDorContext";
+import GlobalState from './context/GlobalState';
+import JSONEditorComponent from "./components/Editor/Editor";
+import TDViewer from './components/TDViewer/TDViewer';
+import './App.css';
+import AppFooter from './components/App/AppFooter';
+import AppHeader from './components/App/AppHeader';
+import { Container, Section, Bar } from '@column-resizer/react';
+import { RefreshCw } from 'react-feather';
+
+import { decompressSharedTd, fetchNbTd, fetchTdFromWot } from './share';
+
+const GlobalStateWrapper = (props) => {
+    return (
+        <GlobalState>
+            <App />
+        </GlobalState>
+    );
+}
+
+// The useEffect hook for checking the URI was called twice somehow.
+// This variable prevents the callback from being executed twice.
+let checkedUrl = false;
+const App = (props) => {
+    const context = useContext(ediTDorContext);
+
+    const [doShowJSON, setDoShowJSON] = useState(false);
+
+    useEffect(() => {
+        if (checkedUrl || (window.location.search.indexOf("td") <= -1 &&
+            window.location.search.indexOf("wotid") <= -1 &&
+            window.location.search.indexOf("localstorage") <= -1)) {
+            return;
+        }
+        checkedUrl = true;
+
+        const url = new URL(window.location.href);
+        const compressedTd = url.searchParams.get("td");
+        if (compressedTd !== null) {
+            const td = decompressSharedTd(compressedTd);
+            if (td === undefined) {
+                alert("The lz compressed TD found in the URL couldn't be reconstructed.");
+                return;
+            };
+
+            context.updateOfflineTD(JSON.stringify(td, null, 2));
+        }
+
+        const nativeTdId = url.searchParams.get("wotid");
```

Comment:
```suggestion
        const nativeTdId = url.searchParams.get("proxyid");
```

also other instances of this should be changed